### PR TITLE
Update Feature Policy JS API

### LINKED ISSUE #4367: document.policy vs document.featurePolicy: What do we do?
[Feature Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy) JS API spec elements were [renamed in December last year](https://github.com/w3c/webappsec-feature-policy/pull/262/commits/919192a1e51de519cd6422e2407a431bc4df040e). Most importantly, `document.policy` was renamed to `document.featurePolicy` and shipped under the new name in Chrome. Firefox still has the old names which are behind a flag and are obviously incompatible with Chrome's.

How should MDN and specifically this repo document this API? Should `document.policy` be marked as 

### Here is the background on this (as of writing):
 - [Spec changes](https://github.com/w3c/webappsec-feature-policy/pull/262/commits/919192a1e51de519cd6422e2407a431bc4df040e):
    - interface `Policy` renamed to `FeaturePolicy`
    - `Document` readonly attribute renamed from `policy` to `featurePolicy` (read `document.policy` renamed to `document.featurePolicy`)
       Please note that this is the breaking change for most developers, unless they specifically account for this difference.
 - [Chrome Platform status](https://www.chromestatus.com/features/5190687460950016) describes `document.policy`, not `document.featurePolicy` (even though [the APIs were renamed in January](https://bugs.chromium.org/p/chromium/issues/detail?id=917057), before enabling by default)
 - MDN:
    - https://developer.mozilla.org/en-US/docs/Web/API/Document/policy
 - this repo:
    - Contains [`api.Document.policy`](https://github.com/mdn/browser-compat-data/blob/master/api/Document.json#L8231) which was not updated yet
  - there are no W3C tests yet, as far as I know

I'll create a PR with `document.featurePolicy` that would contain info about Chrome. But what should we do with `document.policy`?

### PR DESCRIPTION
**Pull request by bershanskiy**
This fixes #4367.

## Summary
Updates `api/Document.json` to rename `Document.policy` into `Document.featurePolicy`, as discussed in #4367.

## Data
Chrome compatibility data sources are described in `notes`. Opera data was extrapolated from Chrome versions and corresponding Opera versions. Opera for Android data not meaningfully updated (just mentioned that the old name) because newer versions are not yet included in `browsers/opera_android.json`.

## Relevant issues
#4367 and every issue mentioned in it and this commit's `notes`.

A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [x] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [x] Data: if you tested something, describe how you tested with details like browser and version
- [x] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [x] Link to related issues or pull requests, if any

### CODE REVIEW

File: api/Document.json [hunk 1]
```diff
@@ -8307,16 +8372,37 @@
...
-            }
+            "webview_android": [
+              {
+                "version_added": "74",
+                "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes it as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
+              },
+              {
+                "version_added": "73",
+                "version_removed": "74",
+                "flags": [
```
Comment (jpmedley) [inline api/Document.json]: Flags can't be turned on for WebView. Please remove.

Comment (bershanskiy) [reply api/Document.json]: Thank you for prompt review!
I removed all Android WebView objects that have flags leaving only the last one. Please note that previously data had Android WebView data with flags and I removed that as well.

**Comment by bershanskiy** [comment]
@jpmedley Did you mean to approve the PR without merging it? Are you waiting for me to update other files (HTMLiFrameElement.json and Policy.json)?

I updated HTMLiFrameElement.json [here on my fork](https://github.com/bershanskiy/browser-compat-data/commit/1cb377ee6a0ed3c8748f85f624c5db5d3c554d12) (which is mostly identical to this PR because turns out the same issues also update iframes even if they do not explicitly mention it). Should I push this change on top of this PR to commit them at the same time or should I make a separate PR?

File: api/Document.json [hunk 2]
```diff
@@ -8228,30 +8228,72 @@
...
-                  "type": "preference",
-                  "name": "#enable-experimental-productivity-features",
-                  "value_to_set": "Enabled"
-                }
-              ]
-            },
+            "chrome": [
+              {
+                "version_added": "74",
+                "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes it as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
```
Comment (vinyldarkscratch) [inline api/Document.json]: We can probably improve the wording on this.
```suggestion
                "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes this variable as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
```

File: api/Document.json [hunk 3]
```diff
@@ -8228,30 +8228,72 @@
...
+                "version_added": "73",
+                "version_removed": "74",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": "#enable-experimental-productivity-features",
+                    "value_to_set": "Enabled"
+                  }
+                ],
+                "notes": "It was renamed from <code>document.policy</code> to <code>document.featurePolicy</code> in <a href='https://crbug.com/917057'>bug 917057</a>."
```
Comment (vinyldarkscratch) [inline api/Document.json]: I think we can leave this note out, as it's already specified by the data.  (I'll only comment on this one, but this applies to all of the notes of the same message.)

File: api/Document.json [hunk 4]
```diff
@@ -8260,39 +8302,62 @@
...
+              "alternative_name": "policy",
               "flags": [
                 {
                   "type": "preference",
                   "name": "dom.security.featurePolicy.webidl.enabled",
                   "value_to_set": "true"
                 }
-              ]
+              ],
+              "notes": "Will eventually be renamed from <code>document.policy</code> to <code>document.featurePolicy</code>, see <a href='https://bugzil.la/1514296'>bug 1514296</a> for details."
```
Comment (vinyldarkscratch) [inline api/Document.json]: I think we can also leave this note out as well?

File: api/Document.json [hunk 5]
```diff
@@ -8228,30 +8228,68 @@
...
+                    "name": "#enable-experimental-productivity-features",
+                    "value_to_set": "Enabled"
+                  }
+                ]
+              }
+            ],
+            "chrome_android": [
+              {
+                "version_added": "74",
+                "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes it as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
```
Comment (vinyldarkscratch) [inline api/Document.json]: Whoops, missed the other two in my last review, sorry!

```suggestion
                "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes this variable as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
```

File: api/Document.json [hunk 6]
```diff
@@ -8308,14 +8367,8 @@
...
-              "version_added": "69",
-              "flags": [
-                {
-                  "type": "preference",
-                  "name": "#enable-experimental-productivity-features",
-                  "value_to_set": "Enabled"
-                }
-              ]
+              "version_added": "74",
+              "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes it as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
```
Comment (vinyldarkscratch) [inline api/Document.json]:
```suggestion
              "notes": "<a href='https://www.chromestatus.com/feature/5190687460950016'>Chrome Platform Status</a> incorrectly describes this variable as <code>document.policy</code>. It was renamed in <a href='https://crbug.com/917057'>bug 917057</a> before being shipped in <a href='https://crbug.com/703703#c22'>bug 703703</a>."
```

**Comment by bershanskiy** [comment]
> Once these changes have been committed, I'll go ahead and merge this right away. 😉

Actually, if it's convenient for you, could you wait for 10-30 more min and I'll do the iframes as well?

**Comment by vinyldarkscratch** [comment]
Oh sure thing, yeah, thank you very much!

**Comment by bershanskiy** [comment]
Iframes are done, but `api.Policy` (`api/Policy.json`) is still not updated. I can commit that as well, if that would make sense to include in this PR.

**Comment by vinyldarkscratch** [comment]
I think that it would make sense to perform that change in this PR as well.  Thanks for double-checking!

**Comment by jpmedley** [comment]
@bershanskiy Yes. Instead of using GitHub's approve option, I should have said in the main conversation that I approved of the Chrome content. I don't know what I was thinking. Generally, I won't approve PRs that touch multiple browsers. I'm not qualified to review browsers other than Chrome. (I'm a Google employee, FYI.)

**Comment by bershanskiy** [comment]
I just missed the [0.0.85 release](https://github.com/mdn/browser-compat-data/tree/v0.0.85) :(, but hopefully this will land in time for the next week's release.

## Data
 - `FeaturePolicy` object and three main methods `allowsFeature`, `allowedFeatures`, `getAllowlistForFeature` are supported by the same sources as the rest of this PR.
### `features` method
 - `features` method was added in Chrome 74 and never was hidden behind a flag. It is not explicitly mentioned anywhere in sources I cited earlier, instead it's mentioned here:
   - the actual [issue](https://bugs.chromium.org/p/chromium/issues/detail?id=917070) fixed by a commit that landed in Chrome 74 (this issue blocked [the main issue](https://bugs.chromium.org/p/chromium/issues/detail?id=703703) that landed in Chrome 74).
   - [_Introduction to Feature Policy_ on Google Developer Blog](https://developers.google.com/web/updates/2018/06/feature-policy)

 - `features` method is not yet implemented in Firefox, see [bug 1514474](https://bugzilla.mozilla.org/show_bug.cgi?id=1514474).

As before, I leave `opera_android` data a bit out of date because `browsers/opera_android.json` is a bit out of date.

**Comment by bershanskiy** [comment]
@vinyldarkscratch I rebased the PR. There were no conflicts to resolve manually, just rm Policy.json and add FeaturePolicy.json.

P.S. I jut saw you added the "rebase-needed" label and rebased right away.

**Comment by vinyldarkscratch** [comment]
Thanks for rebasing!  All in all, this PR looks good to me, and I think we're all ready to go.  Thanks for your hard work regarding this!

**Pull request merged**